File: typescript/packages/bench/src/rules/maximumBacklogRule.ts
/** Cardinality fixture: nominal (class, `implements`) implementor of RuleContract. */

import type { RuleContract } from "../contracts/ruleContract";
import type { Dataset } from "../dataset";

export class MaximumBacklogRule implements RuleContract {
    readonly key = "maximum-backlog";

    evaluate(data: Dataset): boolean {
        return data.openOrders().length >= 1;
    }
}

File: rust/crates/atelier-bench/src/rules/maximum_backlog_rule.rs
//! Rule 2/24 (textual half).

use crate::contracts::rule_contract::RuleContract;
use crate::dataset::Dataset;

pub struct MaximumBacklogRule;

impl RuleContract for MaximumBacklogRule {
    fn evaluate(&self, data: &Dataset) -> bool {
        !data.open_orders().is_empty()
    }
}
